plugins/gstperf.c

```diff
@@ -179,6 +212,21 @@ gst_perf_set_property (GObject * object, guint property_id,
       perf->print_arm_load = g_value_get_boolean (value);
       GST_OBJECT_UNLOCK (perf);
       break;
+    case PROP_BITRATE_WINDOW_SIZE:
+      GST_OBJECT_LOCK (perf);
+      perf->bps_window_size = g_value_get_uint (value);
+      GST_OBJECT_UNLOCK (perf);
+      break;
+    case PROP_BITRATE_INTERVAL:
+      GST_OBJECT_LOCK (perf);
+      perf->bps_interval = g_value_get_uint (value);
+      if (perf->bps_running) {
+        g_source_remove (perf->bps_source_id);
+        perf->bps_source_id =
```

Comment:
This should not be inside the conditional, you always want to install the callback, the conditional only protects the source removal.

Reply:
I don't think I like this approach. In this case, if the user explicitly sets the property at startup, the thread will start (in NULL). And I think we wanted reports only on PAUSED/PLAYING. Plus, the code is starting to look a bit unnecessarily cluttered.